### PR TITLE
fix(platform): add sameFilenameAllowed

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#4474

#### Please provide a brief summary of this pull request.
Added `sameFilenameAllowed` input that allows using the same name for a file when editing the file name.
'Same name' refers to an already existing file name in the list.
Default value: `false`
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [X] Run npm run build-pack-library and test in external application

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Review by olshansky — APPROVED** [review]

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts [hunk 1]
```diff
@@ -292,14 +300,18 @@ export class UploadCollectionComponent {
...
         const itemName = input.value;
-        const newName = item.type === 'file' ? `${itemName}.${item.name.split('.').pop()}` : itemName;
+        const newName = currentItem.type === 'file' ? `${itemName}.${currentItem.name.split('.').pop()}` : itemName;
+
+        if (currentItem.name === newName) {
+            return;
+        }
 
         if (itemName.length > this.maxFilenameLength) {
             const reanmedItem = {
```
Comment (platon-rov) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:313]: Please fix a typo 'reanmedItem` => `renamedItem`

**Review by platon-rov — APPROVED** [review]

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts [hunk 2]
```diff
@@ -156,6 +157,13 @@ export class UploadCollectionComponent {
     @Input()
     showSearch = true;
 
+    /**
+     * Allows to use the same name for a file when editing the file name.
+     * 'Same name' refers to an already existing file name in the list.
+     * */
+    @Input()
+    sameFilenameAllowed = false;
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:165]: Not a big deal but in my opinion it should be named `sameFileNameAllowed` to be consistent with `_fileNameChange`

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Nice work, however one small improvement would be to disable the checkbox while the file name input is invalid.  Also I believe the check should occur when the value in the input changes, not when it is blurred

**Comment by Betrozov** [comment]
Hi @mikerodonnell89 
> Nice work, however one small improvement would be to disable the checkbox while the file name input is invalid.

I cannot disable the checkbox because the Table component does not provide this functionality.

> Also I believe the check should occur when the value in the input changes, not when it is blurred

Will be done

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html [hunk 3]
```diff
@@ -177,16 +177,26 @@
...
+                                            i18n-placeholder
+                                            placeholder="Enter a name"
+                                            compact="true"
+                                            [state]="item.sameFilenameState"
+                                        />
+                                        <span [innerText]="_getItemExtension(item)"></span>
+                                    </div>
+
+                                    <fd-form-message [type]="item.sameFilenameState">
+                                        {{ item.type === 'file' ? 'File' : 'Folder' }} already exists with that name.
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html:197]: i18n?

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts [hunk 4]
```diff
@@ -346,6 +362,24 @@ export class UploadCollectionComponent {
             });
     }
 
+    /** @hidden */
+    _checkName(e: FocusEvent, currentItem: UploadCollectionItem): void {
+        const input = e.target as HTMLInputElement;
+        const itemName = input.value;
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:368]: trim?

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts [hunk 5]
```diff
@@ -292,14 +300,18 @@ export class UploadCollectionComponent {
     /**@hidden
      * Opens the file selector.
      */
-    _fileNameChange(e: FocusEvent, item: UploadCollectionItem): void {
+    _fileNameChange(e: FocusEvent, currentItem: UploadCollectionItem): void {
         const input = e.target as HTMLInputElement;
         const itemName = input.value;
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:305]: trim?

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts (hunk 1 above)
Comment (InnaAtanasova) [reply libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts:313]: Typo still not fixed

**Review by InnaAtanasova — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request closed without merging**